tests/uts/test_agent.py

```diff
@@ -0,0 +1,64 @@
+import json
+import logging
+import os
+import unittest
+import uuid
+from threading import Thread
+
+from fakeredis import FakeStrictRedis
+from mock import patch
+
+from agent.talker import TalkerAgent
+
+
+def get_uuid():
+    return uuid.uuid4().__str__()
+
+
+def kill_process_before_signaling(self, job_id, signal):
+    job = self.current_processes.get(job_id)
+    os.killpg(os.getpgid(job.popen.pid), signal)
+    while job.popen.poll() is None:
+        pass
+    job.send_signal(signal)
+
+
+JOBS_DIR = '/tmp/talker/jobs'
+JOBS_SEEN = '/tmp/talker/jobs/eos.json'
+
+
+@patch('agent.talker.JOBS_DIR', JOBS_DIR)
+@patch('agent.talker.JOBS_SEEN', JOBS_SEEN)
+class TestClient(unittest.TestCase):
+
+    def setUp(self):
+        super(TestClient, self).setUp()
+        if not os.path.isdir(JOBS_DIR):
+            os.makedirs(JOBS_DIR)
+        self.agent = TalkerAgent()
+        self.agent.host_id = get_uuid()
+        self.agent.redis = FakeStrictRedis()
+        self.agent.pending_exception = None
+        self.agent_thread = Thread(target=self.agent.start)
+        self.agent_thread.start()
+
+    def tearDown(self):
+        self.agent.kill()
```

Comment:
you should also join the thread

Reply:
It is a bit problematic. `join` blocks, if we use `join` it means that we wait for the `agent_thread` to finish, this happens only in the `tearDown`. Do you want to put it after `self.agent.kill()`?